numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1810,39 +1831,75 @@ unpack_bits(PyObject *input, int axis, PyObject *count_obj, char order)
         unsigned const char *inptr = PyArray_ITER_DATA(it);
         char *outptr = PyArray_ITER_DATA(ot);
 
-        if (order == 'b') {
-            for (index = 0; index < in_n; index++) {
-                for (i = 0; i < 8; i++) {
-                    *outptr = ((*inptr & (128 >> i)) != 0);
-                    outptr += out_stride;
+        if (out_stride == 1) {
```

Comment:
Aside: If you're feeling adventurous - you might be able to make a similar optimization for `out_stride == -1`

Reply:
I'm not sure that can even happen as the output is allocated by the same function and should not have negative strides
but even if its probably unnecessary to have that little extra in the code, use cases are unlikely